src/App.js

```diff
@@ -1,16 +1,66 @@
-import React from 'react';
+import { React, useState } from 'react';
 import './App.css';
-import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import chatJSON from './data/messages.json';
+import ColorPicker from './components/ColorPicker';
+import ChatDescription from './components/ChatDescription';
 
 const App = () => {
+  // State setup
+  const [chatData, setChatData] = useState(chatJSON.messages);
+  const [numLikes, setNumLikes] = useState(
+    chatJSON.messages.reduce((prev, msg) => {
+      return msg.liked ? prev++ : prev;
+    }, 0)
+  );
+  const [colorClasses, setColorClasses] = useState({
+    local: '🟠',
+    remote: '🟣',
+  });
+  // Event handlers passed down to components
+  const updateMessage = (updatedChat) => {
+    const chats = chatData.map((chatEntry) => {
+      return chatEntry.id === updatedChat.id ? updatedChat : chatEntry;
+    });
+    setChatData(chats);
+    updatedChat.liked ? setNumLikes(numLikes + 1) : setNumLikes(numLikes - 1);
```

Comment:
When state represents something like a counter, we need to depend on a state's previous value and increment the previous value. It's not wrong to do it the way you increment/decrement numLikes, but wanted to call out that we can use prevState too.

`setLikedCount(prevState => prevState + 1)`

This video [here](https://www.youtube.com/watch?v=yvTGXH7uybA) does a great job explaining why